app/models/concerns/orchestration/dhcp.rb

```diff
@@ -93,16 +96,27 @@ def boot_server
 
   private
 
+  def build_dhcp_record(record_mac)
+    raise ::Foreman::Exception.new(N_("DHCP not supported for this NIC")) unless dhcp?
+    record_attrs = dhcp_attrs(record_mac)
+    record_type = (provision? && jumpstart?) ? Net::DHCP::SparcRecord : Net::DHCP::Record
+    handle_validation_errors do
```

Comment:
Nice job extracting this method, it's much more readable and it's easy to see how the record is created now